File: app/src/main/java/in/ceeq/alec/provider/DatabaseUtils.java
package in.ceeq.alec.provider;

import android.content.ContentResolver;
import android.net.Uri;


public class DatabaseUtils {
    public static String getTableContentItemType(String mTableName) {
        return new StringBuilder(ContentResolver.CURSOR_ITEM_BASE_TYPE).append("/").append(mTableName).toString();
    }

    public static String getTableContentType(String mTableName) {
        return new StringBuilder(ContentResolver.CURSOR_DIR_BASE_TYPE).append("/").append(mTableName).toString();
    }

    public static Uri getTableContentUri(String mTableName) {
        return Uri.parse("content://" + ContactContract.AUTHORITY + "/" + mTableName);
    }
}

File: app/src/main/java/in/ceeq/lyte/data/contentprovider/DatabaseUtils.kt
package `in`.ceeq.lyte.data.contentprovider

import android.content.ContentResolver
import android.net.Uri

infix fun String.toContentItemType(tableName: String)
        = "${ContentResolver.CURSOR_ITEM_BASE_TYPE}//$tableName"

infix fun String.toContentType(tableName: String)
        = "${ContentResolver.CURSOR_DIR_BASE_TYPE}//$tableName"

infix fun String.toTableContentUri(tableName: String)
        = Uri.parse("content://${ContactContract.AUTHORITY}//$tableName")
